lib/dependencies/ImportMetaContextDependencyParserPlugin.js

```diff
@@ -0,0 +1,98 @@
+/*
+	MIT License http://www.opensource.org/licenses/mit-license.php
+	Author Ivan Kopeykin @vankop
+*/
+
+"use strict";
+
+const {
+	evaluateToIdentifier
+} = require("../javascript/JavascriptParserHelpers");
+const ImportMetaContextDependency = require("./ImportMetaContextDependency");
+
+/** @typedef {import("estree").Expression} ExpressionNode */
+/** @typedef {import("estree").ObjectExpression} ObjectExpressionNode */
+/** @typedef {import("../javascript/JavascriptParser")} JavascriptParser */
+
+/**
+ * @param {JavascriptParser} parser parser
+ * @param {ObjectExpressionNode} optionsNode node
+ * @returns {{mode: string, recursive: boolean, regExp: RegExp}} options
+ */
+function getOptions(parser, optionsNode) {
+	let regExp = /^\.\/.*$/;
+	let recursive = true;
+	let mode = "sync";
+	if (optionsNode) {
+		for (const prop of optionsNode.properties) {
+			if (prop.type !== "Property" || prop.key.type !== "Identifier") return;
+			switch (prop.key.name) {
+				case "regExp": {
+					const regExpExpr = parser.evaluateExpression(
+						/** @type {ExpressionNode} */ (prop.value)
+					);
+					if (!regExpExpr.isRegExp()) return;
+					regExp = regExpExpr.regExp;
+					break;
+				}
+				case "mode": {
+					const modeExpr = parser.evaluateExpression(
+						/** @type {ExpressionNode} */ (prop.value)
+					);
+					if (!modeExpr.isString()) return;
+					mode = modeExpr.string;
+					break;
+				}
```

Comment:
`mode` need to be handled directly during parsing in that plugin. It can't be passed to the dependency.